arango/responses/lazy.py

```diff
@@ -0,0 +1,32 @@
+from arango.responses import BaseResponse
+
+
+class LazyResponse(BaseResponse):  # pragma: no cover
+    """Response which lazily loads all values from some future.
+
+    :param future: The future instance which has the function result()"
+    :type future: :method:result()
+    :param response_mapper: the callable which maps the result to a standard
+    dictionary of fields
+    :type response_mapper: callable
+    """
+
+    # noinspection PyMissingConstructor
```

Comment:
Yes.  Because if you access any of the fields in the async response it will have to block for the future to be ready.  Instead it waits until one of the fields specified in __slots__ is requested before touching the future.